src/utils.ts

```diff
@@ -128,3 +134,51 @@ export function isValidAmount(amount: number): boolean {
   if (amount <= 0 || !Number.isSafeInteger(amount)) return false;
   return true;
 }
+
+/**
+ * The unblind output function's result interface.
+ */
+export interface UnblindResult {
+  asset: string;
+  value: string;
+}
+
+/**
+ * Unblind an output using confidential.unblindOutput function from liquidjs-lib.
+ * @param output the output to unblind.
+ * @param blindKey the private blinding key.
+ */
+export function unblindOutput(output: Output, blindKey: Buffer): UnblindResult {
+  const result: UnblindResult = { asset: '', value: '' };
+
+  if (!output.rangeProof) {
+    throw new Error('The output does not contain rangeProof.');
+  }
+
+  const unblindedResult: UnblindOutputResult = confidential.unblindOutput(
+    output.nonce,
+    blindKey,
+    output.rangeProof,
+    output.value,
+    output.asset,
+    output.script
+  );
+
+  result.asset = toAssetHash(unblindedResult.asset);
+  result.value = unblindedResult.value;
+  return result;
+}
+
+const emptyNonce: Buffer = Buffer.from('0x00', 'hex');
+
+function bufferNotEmptyOrNull(buffer?: Buffer): boolean {
+  return buffer != null && buffer.length > 0;
+}
+
+/**
+ * Checks if a given output is a confidential one.
+ * @param output the ouput to check.
+ */
+export function isConfidentialOutput(output: Output): boolean {
+  return bufferNotEmptyOrNull(output.rangeProof) && output.nonce !== emptyNonce;
```

Comment:
you should check the surjectionProof as well. (At least is what we so in blindOutputs to determine blinded prevouts)